backend/LexBoxApi/GraphQL/CustomTypes/RefreshJwtOrgMembershipMiddleware.cs

```diff
@@ -0,0 +1,77 @@
+using HotChocolate.Resolvers;
+using LexBoxApi.Auth;
+using LexCore.Auth;
+using LexCore.Entities;
+
+namespace LexBoxApi.GraphQL.CustomTypes;
+
+public class RefreshJwtOrgMembershipMiddleware(FieldDelegate next)
+{
+    private const string REFRESHED_USER_KEY = "RefreshedJwtProjectMembership";
+    // Yes, "Project". We need the same key as in RefreshJwtProjectMembershipMiddleware so that we don't call RefreshUser twice.
+    public async Task InvokeAsync(IMiddlewareContext context)
+    {
+        await next(context);
+        if (UserAlreadyRefreshed(context))
```

Comment:
In OrgGqlConfiguration, I count on the refresh-JWT middleware running before the middleware that handles the `org.Members` field. That way if the user has just clicked the "Leave Org" button, his JWT will contain the newly-updated list of orgs, **minus** the org whose page he's currently on. Which means that inside OrgMembersVisibilityMiddleware I can use `user.Orgs` from the JWT and avoid doing a DB query.

If the refresh were to happen only at the end of the request, I would be forced to use a DB query in OrgMembersVisibilityMiddleware and OrgProjectsVisibilityMiddleware to look up what orgs the user is currently in. So I think this might not be a viable suggestion.